File: src/main/java/greedy/stock_once.go
package main

func main() {

}

func maxProfit(prices []int) int {
	curMin := prices[0]
	res := 0
	for _, val := range prices {
		curMin = min(curMin, val)
		res = max(res, val-curMin)
	}
	return res
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

File: src/main/java/greedy/StockOnce.java
package greedy;

/**
 * @author ShaneTang
 * @create 2021-02-21 22:43
 */
public class StockOnce {

    public int maxProfit(int[] prices) {
        // write code here
        int res = 0;
        int min = Integer.MAX_VALUE; // 不是0
        for (int price : prices) {
            min = Math.min(min, price);
            res = Math.max(res, price - min);
        }
        return res;
    }
}
